.github/workflows/ci.yml

```diff
@@ -0,0 +1,412 @@
+      # Set env vars so PG tools like createdb and dropdb do the right thing.
+      PGHOST: postgres
+      PGUSER: postgres
+      PGPASSWORD: postgres
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3'
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: Build and generate sample data
+        run: |
+          # Generate enclave signing key
+          openssl genrsa -out Enclave_private.pem -3 3072
+          export CONSENSUS_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export INGEST_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export LEDGER_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export VIEW_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export MC_LOG=debug
+
+          # Build binaries
+          cargo build \
+              -p mc-admin-http-gateway \
+              -p mc-consensus-mint-client \
+              -p mc-consensus-service \
+              -p mc-crypto-x509-test-vectors \
+              -p mc-fog-distribution \
+              -p mc-fog-ingest-client \
+              -p mc-fog-ingest-server \
+              -p mc-fog-ledger-server \
+              -p mc-fog-report-server \
+              -p mc-fog-sql-recovery-db \
+              -p mc-fog-test-client \
+              -p mc-fog-view-server \
+              -p mc-ledger-distribution \
+              -p mc-mobilecoind \
+              -p mc-util-generate-sample-ledger \
+              -p mc-util-grpc-admin-tool \
+              -p mc-util-keyfile \
+              -p mc-util-seeded-ed25519-key-gen \
+              --release
+
+          BIN_DIR="$PWD/target/release"
+
+          # Run in temp dir to appease check-dirty-git.
+          mkdir -p /tmp/fog-local-network
+          cd /tmp/fog-local-network
+
+          # Generate sample keys and ledger.
+          FOG_AUTHORITY_ROOT=$("$BIN_DIR/mc-crypto-x509-test-vectors" --type=chain --test-name=ok_rsa_head)
+          "$BIN_DIR/sample-keys" --num 10 --seed=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+          "$BIN_DIR/generate-sample-ledger" --txs 100
+
+          # Generate sample Fog keys.
+          "$BIN_DIR/sample-keys" --num 4 --output-dir fog_keys --fog-report-url 'insecure-fog://localhost:6200' --fog-authority-root $FOG_AUTHORITY_ROOT
+      - name: Run local network
+        run: |
+          BIN_DIR="$PWD/target/release"
+          SCRIPT_DIR="$PWD/tools/fog-local-network"
+
+          cd /tmp/fog-local-network
+
+          # Run local network in background.
+          MC_LOG="debug,rustls=warn,hyper=warn,tokio_reactor=warn,mio=warn,want=warn,rusoto_core=error,h2=error,reqwest=error,rocket=error,<unknown>=error" \
+          LEDGER_BASE="$PWD/ledger" \
+          python3 "$SCRIPT_DIR/fog_local_network.py" --network-type dense5 --skip-build &
```

Comment:
Why are you opting to skip the build here and instead build manually above?

Reply:
Yeah, it's not great. I copied this logic over from `.circleci/config.yml`, which does this logic, I think so it can run both `fog-distribution` and `test_client` against the local network.